core/src/OpenMP/Kokkos_OpenMP_Parallel.hpp

```diff
@@ -379,14 +381,32 @@ class ParallelReduce<FunctorType, Kokkos::RangePolicy<Traits...>, ReducerType,
     const size_t pool_reduce_bytes =
         Analysis::value_size(ReducerConditional::select(m_functor, m_reducer));
 
+    std::lock_guard<std::mutex> lock(m_instance->m_pool_mutex);
+
     m_instance->resize_thread_data(pool_reduce_bytes, 0  // team_reduce_bytes
                                    ,
                                    0  // team_shared_bytes
                                    ,
                                    0  // thread_local_bytes
     );
 
-    const int pool_size = OpenMP::impl_thread_pool_size();
+    if (OpenMP::in_parallel() &&
+        !(omp_get_nested() && (omp_get_level() == 1))) {
```

Comment:
We check this condition a bunch of times.  Did you consider encapsulating it in a function?